lib/encoding/json.toit

```diff
@@ -101,128 +98,15 @@ decode-stream reader:
-    '\b': 'b',
-    '\f': 'f',
-    '\n': 'n',
-    '\r': 'r',
-    '\t': 't',
-    '"':  '"',
-    '\\': '\\'
-}
-
-// A non-zero for every UTF-8 code unit that needs escaping, and a '0' for
-// every code unit that doesn't need escaping.  The number indicates how
-// many extra bytes the escaped version has.
-create-escaped-char-map_ -> ByteArray:
-  // Most control characters (0-31) are output in the form \u00.. which takes 6
-  // characters (5 extra).
-  result := ByteArray 0x100: it < ' ' ? 5 : 0
-  ONE-CHAR-ESCAPES_.do: | code escape | result[code] = 1
-  return result
-
-/**
-Returns a string or a byte array that has been escaped for use in JSON.
-This means that control characters, double quotes and backslashes have
-  been replaced by backslash sequences.
-*/
-escape-string str/string -> any:
-  if str == "" or str.size == 1 and ESCAPED-CHAR-MAP_[str[0]] == 0: return str
-  counter := ByteArray 2
-  bitmap.blit str counter str.size
-      --destination-pixel-stride=0
-      --lookup-table=ESCAPED-CHAR-MAP_
-      --operation=bitmap.ADD-16-LE
-  extra-chars := LITTLE-ENDIAN.uint16 counter 0
-  if extra-chars == 0: return str  // Nothing to escape.
-  if extra-chars == 0xffff:
-    // Overflow of the saturating counter :-(.  We must count manually.
-    extra-chars = 0
-    str.size.repeat: extra-chars += ESCAPED-CHAR-MAP_[str.at --raw it]
-  result := ByteArray str.size + extra-chars
-  output-posn := 0
-  not-yet-copied := 0
-  str.size.repeat: | i |
-    byte := str.at --raw i
-    if ESCAPED-CHAR-MAP_[byte] != 0:
-      result.replace output-posn str not-yet-copied i
-      output-posn += i - not-yet-copied
-      not-yet-copied = i + 1
-      result[output-posn++] = '\\'
-      if ONE-CHAR-ESCAPES_.contains byte:
-        result[output-posn++] = ONE-CHAR-ESCAPES_[byte]
-      else:
-        result[output-posn    ] = 'u'
-        result[output-posn + 1] = '0'
-        result[output-posn + 2] = '0'
-        result[output-posn + 3] = to-lower-case-hex byte >> 4
-        result[output-posn + 4] = to-lower-case-hex byte & 0xf
-        output-posn += 5
-  result.replace output-posn str not-yet-copied str.size
-  return result
-
-class Encoder extends Buffer_:
-  encode obj/any [converter]:
-    if obj is string: encode-string_ obj
-    else if obj is num: encode-number_ obj
-    else if identical obj true: encode-true_
-    else if identical obj false: encode-false_
-    else if identical obj null: encode-null_
-    else if obj is Map: encode-map_ obj converter
-    else if obj is List: encode-list_ obj converter
-    else:
-      result := converter.call obj this
-      if result != null: encode result converter
 
+class Encoder extends EncoderBase_:
+  /** See $EncoderBase_.encode */
   encode obj/any converter/Lambda:
-    encode obj: converter.call obj this
+    return super obj converter
 
-  encode obj/any:
-    encode obj: throw "INVALID_JSON_OBJECT"
+  /** See $Buffer_.put-unquoted */
+  put-unquoted data -> none:
+    super data
```

Comment:
delete whet? the comment or the method? The toitdoc compiler was complaining of unresolved referencing, as these are being referenced from json.toit and yaml.toit.

Reply:
I see. That looks like a bug in the compiler. It should find the superclass's method when referencing a method. Otherwise, a benign change can lead to toitdoc errors.
Maybe add a TODO that these methods should be removed once the compiler has been fixed.